nni/compression/pytorch/compressor.py

```diff
@@ -319,8 +318,8 @@ class Pruner(Compressor):
 
     def __init__(self, model, config_list, optimizer=None):
         super().__init__(model, config_list, optimizer)
-        if optimizer is not None:
-            self.patch_optimizer(self.update_mask)
+        # if optimizer is not None:
+        #     self.patch_optimizer(self.update_mask)
```

Comment:
where is `self.update_mask` placed in this version?

Reply:
there is logic changes if this patch is put at the send of `compress` function

Reply:
This mainly affects pruners which handle optimizer, i.e. `TaylorFOWeightFilterPruner`.
In the previous version, `update_mask` happened at the beginning of the finetune stage (outside `compress()`) after the first `optimizer.step()`.
Now, `update_mask` happened after a sparsifying training iteration (inside `compress()`).